test/infrastructure/inmemory/internal/controllers/inmemorymachine_controller.go

```diff
@@ -494,62 +793,115 @@ func (r *InMemoryMachineReconciler) reconcileDelete(ctx context.Context, cluster
 		return ctrl.Result{}, errors.Wrapf(err, "failed to delete Node")
 	}
 
-	if util.IsControlPlaneMachine(machine) {
-		controllerManagerPod := &corev1.Pod{
-			ObjectMeta: metav1.ObjectMeta{
-				Namespace: metav1.NamespaceSystem,
-				Name:      fmt.Sprintf("kube-controller-manager-%s", inMemoryMachine.Name),
-			},
-		}
-		if err := cloudClient.Delete(ctx, controllerManagerPod); err != nil && !apierrors.IsNotFound(err) {
-			return ctrl.Result{}, errors.Wrapf(err, "failed to controller manager Pod")
-		}
+	return ctrl.Result{}, nil
+}
 
-		schedulerPod := &corev1.Pod{
-			ObjectMeta: metav1.ObjectMeta{
-				Namespace: metav1.NamespaceSystem,
-				Name:      fmt.Sprintf("kube-scheduler-%s", inMemoryMachine.Name),
-			},
-		}
-		if err := cloudClient.Delete(ctx, schedulerPod); err != nil && !apierrors.IsNotFound(err) {
-			return ctrl.Result{}, errors.Wrapf(err, "failed to scheduler Pod")
-		}
+func (r *InMemoryMachineReconciler) reconcileDeleteETCD(ctx context.Context, cluster *clusterv1.Cluster, machine *clusterv1.Machine, inMemoryMachine *infrav1.InMemoryMachine) (ctrl.Result, error) {
+	// No-op if the machine is not a control plane machine.
+	if !util.IsControlPlaneMachine(machine) {
+		return ctrl.Result{}, nil
+	}
 
-		apiServer := fmt.Sprintf("kube-apiserver-%s", inMemoryMachine.Name)
-		apiServerPod := &corev1.Pod{
-			ObjectMeta: metav1.ObjectMeta{
-				Namespace: metav1.NamespaceSystem,
-				Name:      apiServer,
-			},
-		}
-		if err := cloudClient.Delete(ctx, apiServerPod); err != nil && !apierrors.IsNotFound(err) {
-			return ctrl.Result{}, errors.Wrapf(err, "failed to apiServer Pod")
-		}
-		if err := r.APIServerMux.DeleteAPIServer(resourceGroup, apiServer); err != nil {
-			return ctrl.Result{}, err
-		}
+	// Compute the resource group unique name.
+	// NOTE: We are using reconcilerGroup also as a name for the listener for sake of simplicity.
+	resourceGroup := klog.KObj(cluster).String()
+	cloudClient := r.CloudManager.GetResourceGroup(resourceGroup).GetClient()
 
-		// TODO: if all the API server are gone, cleanup all the k8s objects from the resource group.
-		// note: it is not possible to delete the resource group, because cloud resources should be preserved.
-		// given that, in order to implement this it is required to find a way to identify all the k8s resources (might be via gvk);
-		// also, deletion must happen suddently, without respecting finalizers or owner references links.
+	etcdMember := fmt.Sprintf("etcd-%s", inMemoryMachine.Name)
+	etcdPod := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: metav1.NamespaceSystem,
+			Name:      etcdMember,
+		},
+	}
+	if err := cloudClient.Delete(ctx, etcdPod); err != nil && !apierrors.IsNotFound(err) {
+		return ctrl.Result{}, errors.Wrapf(err, "failed to etcd Pod")
```

Comment:
```suggestion
		return ctrl.Result{}, errors.Wrapf(err, "failed to delete etcd Pod")
```
same for apiserver below